### DISCUSSION TITLE
Validate JWT token for use in serverless function

### DISCUSSION THREAD

**Discussion by charbs-io** [pocketbase/pocketbase#1933] [Q&A]
I'm integrating Stripe into a SaaS MVP. 

I'm using Netlify functions (serverless functions) to fetch products etc. 

When it comes to more sensitive data, like a customer's subscription for example, it's crucial to lock the function from public use (the netlify functions _are_ publicaly callable).

When I invokve something like `/api/getStripeSubscription`, I'm sending the `Authorization` header with the request. I can then retrieve the token in the function and validate it. 

Although i have no idea how to validate it. 

I'm thinking maybe use something like [https://github.com/auth0/node-jsonwebtoken](https://github.com/auth0/node-jsonwebtoken) - but for that i would need a JWT secret.

Can I perhaps authenticate as a pocketbase admin in the function and then somehow use that to validate the request token?

**Comment by ganigeorgiev** [comment]
We don't have a dedicated API endpoint for token validations, but if you have access to the token in the Netlify function, then you can send an `authRefresh()` call (aka. `POST /api/collections/users/auth-refresh`). If the token is invalid or expired you'll gen an error, otherwise - a new token and up-to-date user data.

```js
// load the token from your custom function
await pb.authStore.save(token, null);

try {
    await pb.collection("users").authRefresh();
} catch (err) {
    console.log("invalid or expired token")
}
```

**Reply by charbs-io** [reply]
Wow - I had no idea you could load a token like this. This worked for my usecase perfectly. Thanks.

**Comment by MarNeumann** [comment]
Thank you so much!
